std-bits/table/src/main/java/org/enso/table/data/index/MultiValueIndex.java

```diff
@@ -93,6 +96,80 @@ public Table makeTable(Aggregator[] columns) {
         merged);
   }
 
+  public Table makeCrossTabTable(
+      Column[] groupingColumns, Column nameColumn, Aggregator[] aggregates) {
+    final int size = locs.size();
+
+    var nameIndex =
+        new MultiValueIndex(new Column[] {nameColumn}, nameColumn.getSize(), null, this.comparator);
+    final int columnCount = groupingColumns.length + nameIndex.locs.size() * aggregates.length;
+
+    // Create the storage
+    Builder[] storage = new Builder[columnCount];
+    IntStream.range(0, groupingColumns.length)
+        .forEach(
+            i -> storage[i] = Builder.getForType(groupingColumns[i].getStorage().getType(), size));
+    IntStream.range(0, nameIndex.locs.size())
+        .forEach(
+            i -> {
+              int offset = groupingColumns.length + i * aggregates.length;
+              IntStream.range(0, aggregates.length)
+                  .forEach(
+                      j -> storage[offset + j] = Builder.getForType(aggregates[j].getType(), size));
+            });
```

Comment:
Any reason for using streams here instead of regular for loops?

(Not saying not to, but in similar code I'm writing I'm using for-loops and so I want to clarify what is our desired style)